### ISSUE TITLE
[Feature Suggestion] Allow chaining image processing ops

### ISSUE THREAD

**Issue by rootkea** [getzola/zola#1507]
Hello!

I'm not sure whether this fits within the scope of Zola or not but how about allowing multiple consecutive image processing ops on an image? e.g. Scale this image and then blur or vice-versa

When other image processing effects get added in future (e..g `blur_image`), such an arrangement will be very helpful.

This only needs two changes:
1. `path` argument to `resize_image`/`blur_image` image processing functions should be relative to PROJECT_ROOT instead of `content`
2. `resize_image`/`blur_image` image processing functions should return output file path relative to `static` dir beginning with `/`. e.g. `/processed_images/bcca10e871e47d2100.jpg` so that they can be used in `<img src=` directly without needing any change. And at the same time can be passed to `path` argument of next image processing function by just appending `"static"` string to it.

Sample example:
```
{% set scaled_img = resize_image(path="content/first/sunset.jpg", width=700, height=150, op="scale") %}
<img src="{{ blur_image(path="static" ~ scaled_img, ..., op="blur") }}" />
```

**Comment by Keats** [comment]
cc @vojtechkral 
I think it's a bit out of scope for Zola but let's see

**Comment by vojtechkral** [comment]
Hmm... current implementation can't handle chained ops as that requires tracking dependencies between ops. Currently we just perform them in arbitrary order / in parallel.
Perhaps another way to do this would be via some kind of multiple-step call, such as
```
image_ops(ops=[
 ["resize_image", "content/first/sunset.jpg", 700, 150, "scale" ],
 ["blur_image", "static" ~ scaled_img, ..., "blur"],
])
```

Buuut... I'm not sure. We probably shouldn't reimplement `imagemagick` as part of Zola...

**Comment by Keats** [comment]
> We probably shouldn't reimplement imagemagick as part of Zola...

Yep, that's my thought as well.

**Comment by rootkea** [comment]
Thanks for the replies guys! I agree with you that this is perhaps outside the scope of Zola indeed. So closing this one!

**Comment by vojtechkral** [comment]
One thought I had was that things like this could be done with some kind of post- or pre-processing support (I do [something like that](https://github.com/vojtechkral/bard/blob/main/doc/bard.toml.md#post-processing) in my _bard_ project), but I'm not sure if that would be in scope either and you can probably do that sort of thing with a Makefile or such. Also this kind of crosses into the area of plugins which is tricky for static site generators to get right :slightly_smiling_face:

**Comment by Keats** [comment]
> you can probably do that sort of thing with a Makefile or such

yep, or a package.json if you already have some front-end dependencies. I don't think it's something that adds a lot of value.
